_posts/2023-09-25-Debezium-JMX-signaling-and-notifications.adoc

```diff
@@ -0,0 +1,1558 @@
+ ✔ Container deploy-postgres-1   Started                                                                                                                                                                                           0.1s
+ ✔ Container deploy-kafka-1      Started                                                                                                                                                                                           0.1s
+ ✔ Container deploy-connect-1    Started
+----
+
+Now we can check that all services are up and running executing the command
+[source, bash]
+----
+docker ps
+----
+
+the output should be something similar to this
+[source, bash]
+----
+CONTAINER ID   IMAGE                            COMMAND                  CREATED         STATUS         PORTS                                                                              NAMES
+f1d49fb79dba   quay.io/debezium/connect:2.4                "/docker-entrypoint.…"   3 seconds ago   Up 2 seconds   0.0.0.0:8083->8083/tcp, 0.0.0.0:8778->8778/tcp, 0.0.0.0:9012->9012/tcp, 9092/tcp   deploy-connect-1
+e164b2651fbf   quay.io/debezium/kafka:2.4       "/docker-entrypoint.…"   3 seconds ago   Up 2 seconds   0.0.0.0:9092->9092/tcp                                                             deploy-kafka-1
+e61116f22f9d   quay.io/debezium/example-postgres:2.4    "docker-entrypoint.s…"   4 seconds ago   Up 2 seconds   0.0.0.0:5432->5432/tcp                                                             deploy-postgres-1
+ccb502882928   quay.io/debezium/zookeeper:2.4   "/docker-entrypoint.…"   4 seconds ago   Up 2 seconds   0.0.0.0:2181->2181/tcp, 0.0.0.0:2888->2888/tcp, 0.0.0.0:3888->3888/tcp             deploy-zookeeper-1
+----
+
+At this point all services are up and running, so we can register the connector through the following configuration
+
+[source, json]
+----
+{
+  "name": "inventory-connector",
+  "config": {
+    "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+    "tasks.max": "1",
+    "database.hostname": "postgres",
+    "database.port": "5432",
+    "database.user": "postgres",
+    "database.password": "postgres",
+    "database.server.id": "184054",
+    "database.dbname": "postgres",
+    "topic.prefix": "dbserver1",
+    "snapshot.mode": "NEVER",
+    "schema.history.internal.kafka.bootstrap.servers": "kafka:9092",
+    "schema.history.internal.kafka.topic": "schema-changes.inventory",
+    "signal.enabled.channels": "source,jmx", <1>
+    "signal.data.collection": "inventory.debezium_signal", <2>
+    "notification.enabled.channels": "jmx"
+  }
+}
+----
+<1> This configuration enables **source** and **jmx** channel. Even if we will use JMX to send the signal, since we want to execute an incremental snapshot, the **source** signal is still required because Debezium needs to use signal table to watermark the db log for events decoupling.
```

Comment:
```suggestion
<1> This configuration enables the **source** and **jmx** channels. Even if we only want to use JMX to send signals to execute an incremental snapshot, the **source** signal is still required because Debezium needs to use signal table to watermark the db log for event deduplication.
```